### DISCUSSION TITLE
13.0.0

### DISCUSSION THREAD

**Discussion by oliemansm** [graphql-java-kickstart/graphql-spring-boot#851] [Ideas]
## Bug fixes
* disable async timeout by default #778
* graphiql config headers do not work in new release 12.0.0 #726
* CorsEnabled=false not working for Spring-Boot-Service-Configs #682
* memory leak due to ReactiveApolloSubscriptionSession not closing #740
* Cannot access AsyncServletProperties.Threads fields because it is a package private class #748
* Graphiql static resource loading fix #794
* quote regex input string #824

## Features
* aliased scalars #659
* Function for GraphQLTestTemplate to upload files using Upload scalar #712
* added ability to use strings for GraphQl queries when using GraphQLTestTemplate #807

## Dependencies
* update dependency com.graphql-java:graphql-java to v18.3 #846
* update lib_graphql_servlet_ver to v13.0.0 #848
* update dependency com.graphql-java-kickstart:graphql-java-tools to v12.1.0 #825
* update spring boot to v2.7.2 #845
* update Reflections to 0.10.2 #772

<hr /><em>This discussion was created from the release <a href='https://github.com/graphql-java-kickstart/graphql-spring-boot/releases/tag/v13.0.0'>13.0.0</a>.</em>

**Comment by 4ig** [comment]
Thank you @oliemansm!
Is there a migration guide from 12.0 to 13.0? I see that a bunch of classes have been removed or deprecated, but it is not clear how to solve these issues.

For instance `DefaultGraphQLServletContext` has been deprecated, and the deprecation note suggests using `GraphQLKickstartContext`, but it is an interface and it is not clear how I can replace the old class with it.

**Reply by oliemansm** [reply]
I need to find some time to write an article for our documentation site on this.

That `GraphQLKickstartContext` remark is at the top level of the class indeed. It's been quite a while since I implemented this tbh, and it could very well be that this was an intermediate step. Thinking about this now it might very well be that this custom context class might be removed entirely in a future release and we'll stay with only the default context as provided by graphql-java.

If you just want to use the options the default context provides, i.e. grab the request or response, then it's recommended to use the context as intended by the graphql-java guys. Meaning that you grab the context from the environment and get objects/values from it by their keys. For example to get the request object:
```java
public void someQuery(DataFetchingEnvironment environment) {
  HttpServletRequest request = environment.getGraphQlContext().get(HttpServletRequest.class);
}
```
The objects that used to be set in the context classes that are now deprecated are all made available this way in this new default context.

The context should be the only thing you'd run into during migration, and even then since it's deprecated right now and not yet removed your current code should still work after upgrading to v13.

**Reply by arjun-oodle** [reply]
Perfect. You saved my day. `environment.getGraphQlContext().get(HttpServletRequest.class);` This is what I was looking for. ❤️
